lib/src/update.dart

```diff
@@ -19,41 +19,54 @@ import 'package:yaml/yaml.dart';
 import 'package:abide/src/constants.dart';
 import 'package:abide/src/util.dart';
 
-Future<Null> updateAnalysisOption(YamlMap abideYaml,
-    {bool uncommentClean}) async {
-  final YamlMap currentAnalysisOptions =
-      loadAnalysisOptions(renameDeprecatedFilename: true);
-  final String currentAnalysisOptionsString = loadAnalysisOptionsAsString();
+Future<String> updateAnalysisOption(YamlMap abideYaml,
+    {String pathToAnalysisOptionsFile,
+    bool uncommentClean,
+    bool writeToFile = true}) async {
+  YamlMap currentAnalysisOptions = loadAnalysisOptions(
+      pathToAnalysisOptionsFile: pathToAnalysisOptionsFile,
+      renameDeprecatedFilename: true);
+
+  final String currentAnalysisOptionsString = loadAnalysisOptionsAsString(
+      pathToAnalysisOptionsFile: pathToAnalysisOptionsFile);
+
   Map<String, Map<String, int>> lintErrorCounts = <String, Map<String, int>>{};
   // Write a version with ALL lint rules enabled
   // and then run dart analyzer to get counts of lint per rule
   // those counts are passed into the 2nd write run to decide
   // whether or not to comment out a lint rule with a lot of
-  // new lint warnings
-  writeAnalyisOptionsFile(
-      all: true,
-      abideYaml: abideYaml,
-      currentAnalysisOptions: currentAnalysisOptions,
-      currentAnalysisOptionsString: currentAnalysisOptionsString);
-  lintErrorCounts = await getLintErrorCounts(abideYaml: abideYaml);
-  writeAnalyisOptionsFile(
+  // new lint warnings. If we're not writing to a file, then lint
+  // counts won't work, so skip that step.
+  if (writeToFile) {
+    writeAnalyisOptionsFile(
+        all: true,
+        abideYaml: abideYaml,
+        currentAnalysisOptions: currentAnalysisOptions,
+        currentAnalysisOptionsString: currentAnalysisOptionsString);
+    lintErrorCounts = await getLintErrorCounts(abideYaml: abideYaml);
+  }
+  return writeAnalyisOptionsFile(
```

Comment:
Kinda weird that it runs this function twice. Would it make more sense to return an object that reflects the changes to be made and then have a separate function that writes them to disk? That would also keep the filesystem code paths separate from the more testable non-filesystem code paths.

Reply:
It always ran twice before. The `getLintErrorCounts` runs the command line dart analyzer to get the number of lints for each rule. That tool needs the analysis options file written to disk. So the first run writes a file with ALL the rules enabled to collect counts for all the rules. Then the 2nd writes it with some commented out due to the recommendations or existing lints hit.
The change here was to not do the first run while testing because we're not writing to disk. (The writeToFile option is only for testing)